appscale/tools/appscale_stats.py

```diff
@@ -0,0 +1,657 @@
+        proc["monit_name"],
+        memory_unique / 1024 / 1024,  # row[2] from B to MB
+        cpu_percent
+      ])
+
+  return process_stats_headers, process_stats
+
+
+def get_summary_process_stats(raw_process_stats, raw_node_stats):
+  """
+  Obtains useful information from summary process statistics and returns:
+  SERVICE (ID), INSTANCES, UNIQUE MEMORY SUM (MB),
+  CPU SUM (%), CPU PER 1 PROCESS (%), CPU PER 1 CORE (%) values.
+
+  Args:
+    raw_process_stats: A dist in which each key is an ip and value is a dict
+      of useful information.
+
+  Returns:
+    A list of summary process statistics headers.
+    A list of summary process statistics.
+  """
+  cpu_count = sum(
+    int(node["cpu"]["count"])
+    for node in raw_node_stats.itervalues()
+  )
+
+  process_stats = []
+  process_list = []
+  unique_processes = set()
+
+  process_stats_headers = [
+    "SERVICE (ID)", "INSTANCES", "UNIQUE MEMORY SUM (MB)",
+    "CPU SUM (%)", "CPU PER 1 PROCESS (%)", "CPU PER 1 CORE (%)"
+  ]
+
+  for proc in raw_process_stats.itervalues():
+    process_list += proc["processes_stats"]
+
+  for proc in process_list:
+    name = proc["unified_service_name"]
+    app_id = proc["application_id"]
+    name_id = name + (" ({})".format(app_id) if app_id else "")
+    if name_id in unique_processes:  # if calculating with process has been made
+      continue
+
+    process_group = [
+      p for p in process_list
+      if p["unified_service_name"] == name
+      and p["application_id"] == app_id
+    ]
+
+    unique_processes.add(name_id)
+    unique_memory = 0
+    cpu_percent = 0
+
+    for p in process_group:
+      unique_memory += int(p["memory"]["unique"])
+      cpu_percent += float(p["cpu"]["percent"])
+      if p["children_num"] > 0:
+        child = p["children_stats_sum"]
+        unique_memory += int(child["memory"]["unique"])
+        cpu_percent += float(child["cpu"]["percent"])
+
+    avg_cpu_per_process = cpu_percent / len(process_group)
+    cpu_percent_cpu_count = cpu_percent / cpu_count
+
+    process_stats.append([
+      name_id,
+      len(process_group),
+      unique_memory / 1024 / 1024,  # row[2] from B to MB
+      cpu_percent,
+      avg_cpu_per_process,
+      cpu_percent_cpu_count
+    ])
+
+  return process_stats_headers, process_stats
+
+
+def get_proxy_stats(raw_proxy_stats, verbose, apps_filter):
+  """
+  Obtains useful information from proxy statistics and returns:
+  SERVICE (ID), UNIQUE MEMORY SUM (MB), CPU PER 1 PROCESS (%),
+  CPU PER 1 CORE (%), CPU SUM (%) values.
+
+  Args:
+    raw_proxy_stats: A dist in which each key is an ip and value is a dict
+      of useful information.
+    verbose: A boolean - verbose or not verbose mode.
+    apps_filter: A boolean - show all services or applications only.
+
+  Returns:
+    A list of proxy statistics headers.
+    A list of proxy statistics.
+  """
+  proxy_stats_headers = [
+    "SERVICE | ID", "SERVERS", "REQ RATE / REQ TOTAL", "5xx / 4xx", "QUEUE CUR"
```

Comment:
"REQ RATE / REQ TOTAL" is an unusual way of displaying these values because it implies that it is a fraction.